packages/common-all/src/utils.ts

```diff
@@ -128,3 +129,24 @@ export class DefaultMap<K, V> {
     return this._internalMap.delete(key);
   }
 }
+
+export class TagUtils {
+  /** Removes `oldTag` from the frontmatter tags of `note` and replaces it with `newTag`, if any. */
+  static replaceTag({note, oldTag, newTag}: {note: NoteProps, oldTag: string, newTag?: string}) {
+    if (_.isUndefined(note.tags) || _.isString(note.tags)) {
+      note.tags = newTag;
+    } else {
+      const index = _.findIndex(note.tags, (tag) => tag === oldTag);
+      if (newTag) {
+        if (index >= 0) {
+          note.tags[index] = newTag;
+        } else {
+          // Weird, can't find the old tag. Add the new one anyway.
```

Comment:
does this happen in practice?

Reply:
No, the old tag always gets removed in the tests. I decided to cover that case just in case.